pkg/apiserver/api_installer.go

```diff
@@ -401,6 +321,140 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage RESTStorage
 	return nil
 }
 
+// rootScopeNaming reads only names from a request and ignores namespaces
+type rootScopeNaming struct {
+	scope meta.RESTScope
+	runtime.SelfLinker
+	itemPath string
+}
+
+func (n rootScopeNaming) Namespace(req *restful.Request) (namespace string, err error) {
+	return
+}
+
+func (n rootScopeNaming) Name(req *restful.Request) (namespace, name string, err error) {
+	name = req.PathParameter("name")
+	if len(name) == 0 {
+		err = errEmptyName
```

Comment:
I do, however, like having names to the two return types rather than just returning (string, string) so that it's obvious to callers what is being returned.